hack/update-deps.sh

```diff
@@ -25,4 +25,7 @@ cd ${REPO_ROOT_DIR}
 # Ensure we have everything we need under vendor/
 dep ensure
 
+rm -rf $(find vendor/ -name 'OWNERS')
+rm -rf $(find vendor/ -name '*_test.go')
```

Comment:
This shouldn't be required :thinking: (`dep` is able prunes tests in recent version — >= 0.4.x for sure)

Reply:
Oddly enough it didn't work for me

```
dep:
 version     : v0.5.0
 build date  : 2018-07-26
 git hash    : 224a564
 go version  : go1.10.3
 go compiler : gc
 platform    : darwin/amd64
 features    : ImportDuringSolve=false
```

Reply:
@dprotaso hum that is really weird then :thinking: :sweat:

Reply:
Prune settings are off on a few dependent packages - that's why they appear

Reply:
https://github.com/knative/build/blob/277d5bca5f217f4c34803e5b1d35ac2f98b228b7/Gopkg.toml#L58-L70

Reply:
That said…

```
dep: WARNING: redundant prune option "go-tests" set for "k8s.io/code-generator"
dep: WARNING: redundant prune option "go-tests" set for "github.com/knative/caching"
dep: WARNING: redundant prune option "go-tests" set for "github.com/knative/test-infra"
```

So there should be no reason for `dep` to keep them…

Reply:
@dprotaso for test-infra, `non-go` is false because we need the shell scripts. We don't care about tests. I agree that there must be a reason for dep to keep some tests, since by default they are pruned.